### PR TITLE
Use threadpool to download

### PR DESCRIPTION
**Pull request by Chiiizzzy**
Signed-off-by: Kaiyuan Hu <kaiyuan.hu@zilliz.com>

### CODE REVIEW

File: towhee/hub/repo_manager.py [hunk 1]
```diff
@@ -275,6 +278,64 @@ def get_file_list(self, author: str, repo: str, commit: str) -> List[str]:
 
         return file_list
 
+    def download_executor(self, author: str, repo: str, tag: str, file_name: str, lfs_files: Tuple[str], local_dir: Union[str, Path]) -> bool:
```
Comment (junjiejiangjjj) [inline towhee/hub/repo_manager.py:281]: Why need `repo`? The `RepoManger` already has the repo info

Comment (Chiiizzzy) [reply towhee/hub/repo_manager.py:281]: We might download other repo, for example we need to download pipeline-template when initialize a new pipeline repo.

File: towhee/hub/repo_manager.py [hunk 2]
```diff
@@ -275,6 +278,64 @@ def get_file_list(self, author: str, repo: str, commit: str) -> List[str]:
...
+            except FileExistsError:
+                pass
+            except OSError as e:
+                raise e
+        # Get content.
+        try:
+            r = requests.get(url, stream=True)
+            if r.status_code == 429:
+                time.sleep(3)
+                self.download_executor(author, repo, tag, file_name, lfs_files, local_dir)
```
Comment (junjiejiangjjj) [inline towhee/hub/repo_manager.py:324]: We may use `for-loop` to retry, not use recursive

**Pull request merged**